Spelling docs

jsoref opened the pull request:
This is the docs portion of #37035 split per @tvernum.

Since there were so few `i.e.`s here, I just left them, it isn't worth dropping, and for docs it seems more useful to leave them fixed.

There are probably merge conflicts, but realistically, as noted in the other PR, there will always be merge conflicts, this repo moves way too fast for large changes like this. I'll cross that bridge closer to when reviews are done.

--
Generated by https://github.com/jsoref/spelling `f`; to maintain your repo, please consider `fchurn`

jsoref commented:
Changes from the original PR:
1. non docs/ commits were dropped (if people were watching, the way I did that was by deleting everything outside of docs/ and rebasing onto that, and then I forgot to remove that commit...)
1. `height` is now `high` per https://github.com/elastic/elasticsearch/pull/37035#discussion_r244551558
1. I dropped `pluggable-similarities.html` per https://github.com/elastic/elasticsearch/pull/37035#discussion_r244551629

elasticmachine commented:
Pinging @elastic/es-docs

javanna commented on x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/attachment/HttpEmailAttachmentParser.java:
```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```
I think you need to remove the corresponding file with the misspelled class name? Possibly you can move the file with git so that we don't lose history around it? You can also do this in a separate smaller PR.

jsoref replied:
*sigh*, I forgot that converting hg moves to git sometimes results in this mess, I'll fix.

javanna commented on x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/notification/email/attachment/HttpEmailAttachmentParserTests.java:
```diff
@@ -0,0 +1,152 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class HttpEmailAttachmentParserTests extends ESTestCase {
```
Same as above, I think you need to remove the corresponding file with the misspelled class name? Possibly you can move the file with git so that we don't lose history around it? You can also do this in a separate smaller PR.

javanna commented:
test this please

javanna commented:
retest this please

javanna commented:
I took the liberty to merge master in, as there were a couple of noisy tests causing failures that got muted on master meanwhile. Remember to pull from your own remote ;)

jsoref commented on docs/java-rest/high-level/security/get-users.asciidoc:
```diff
@@ -2,7 +2,7 @@
 --
 :api: get-users
 :request: GetUsersRequest
-:respnse: GetUsersResponse
```
fwiw, I've written Travis integration for checkstyle: https://github.com/checkstyle/checkstyle/blob/master/.ci/test-spelling-unknown-words.sh

The same sort of thing can be done in any project. If you're not a fan of shell scripts, you can of course port the code to Java/Groovy.

javanna replied:
checking the speeling would work, but in this specific case, given that `:request:` and `:response:` as well as `:api:` are some kind of keyword needed to fill the docs content, we could fail the build when anything other than the known words are used within those delimiters. I think :)

jsoref commented:
I've rebased forward and cleaned out the files that were supposed to be renamed (my original hg repo properly had renames, but somewhere along the way the delete got lost)

javanna commented:
retest this please

jsoref commented:
Yeah, those commits aren't supposed to be there, I'm trying to figure out what I did wrong...

jsoref commented on docs/src/test/java/org/elasticsearch/smoketest/DocsClientYamlTestSuiteIT.java:
```diff
@@ -66,8 +66,8 @@ public DocsClientYamlTestSuiteIT(@Name("yaml") ClientYamlTestCandidate testCandi
                 new ParseField("compare_analyzers"), CompareAnalyzers::parse));
-        NamedXContentRegistry executeableSectionRegistry = new NamedXContentRegistry(entries);
-        return ESClientYamlSuiteTestCase.createParameters(executeableSectionRegistry);
+        NamedXContentRegistry executableSectionRegistry = new NamedXContentRegistry(entries);
```
This is now the only file that's a `.java` -- it's a bit odd that it lives in `docs/`

javanna replied:
Got it, I think it's there because it's the test class that tests the docs snippets. I think it makes sense for it to be there. This way when running ./gradlew :docs:check that one gets run.

jsoref commented:
@javanna : hopefully this should pass

javanna commented:
retest this please

javanna commented:
run gradle build tests 2

javanna commented:
retest this please

lcawl commented on x-pack/docs/en/watcher/release-notes.asciidoc:
```diff
@@ -121,7 +121,7 @@ March 30, 2016
 * Added <<actions-pagerduty, PagerDuty action>>
 * Added support for adding <<configuring-email-attachments, attachments to emails>>
-  via HTTP requests and superceding and deprecating the usage of `attach_data`
+  via HTTP requests and superseding and deprecating the usage of `attach_data`
```
I wouldn't consider this one wrong, but otherwise the changes LGTM

javanna commented:
run gradle build tests 2

javanna commented:
run gradle build tests 1

javanna commented:
retest this please

javanna commented:
thanks a lot @jsoref for all these fixes !!! And even more for making us fix the 5 different ways we were able to misspell the name of our own product :)

jsoref commented:
Fwiw, that isn't uncommon :-o
